spec/server/functional/oauth.spec.coffee

```diff
@@ -0,0 +1,72 @@
+OAuthProvider = require '../../../server/models/OAuthProvider'
+request = require '../request'
+utils = require '../utils'
+
+describe 'POST /db/o-auth', ->
+  url = utils.getURL('/db/o-auth')
+  json = { name: '3rd party' }
+  beforeEach utils.wrap ->
+    yield utils.clearModels([OAuthProvider])
+
+  it 'allows admin to create new OAuthProvider', utils.wrap ->
+    user = yield utils.initAdmin()
+    yield utils.loginUser(user)
+    [res, body] = yield request.postAsync({ url, json })
+    expect(res.statusCode).toBe(201)
+    expect(res.body.name).toEqual(json.name)
+    expect(res.body.creator).toEqual(user.id)
+
+  it 'allows licensor to create new OAuthProvider', utils.wrap ->
+    user = yield utils.initLicensor()
+    yield utils.loginUser(user)
+    [res, body] = yield request.postAsync({ url, json })
+    expect(res.statusCode).toBe(201)
+    expect(res.body.name).toEqual(json.name)
+    expect(res.body.creator).toEqual(user.id)
+
+  it 'returns 403 for normal users', utils.wrap ->
+    user = yield utils.initUser()
+    yield utils.loginUser(user)
+    [res, body] = yield request.postAsync({ url, json })
+    expect(res.statusCode).toBe(403)
+
+describe 'PUT /db/o-auth', ->
+  url = utils.getURL('/db/o-auth')
+  json = { name: '3rd party' }
+  beforeEach utils.wrap ->
+    yield utils.clearModels([OAuthProvider])
+
+  it 'allows licensor to update an OAuthProvider', utils.wrap ->
+    user = yield utils.initLicensor()
+    yield utils.loginUser(user)
+    [res, body] = yield request.postAsync({ url, json })
+    expect(res.statusCode).toBe(201)
+    expect(res.body.name).toEqual(json.name)
+    expect(res.body.creator).toEqual(user.id)
+
+    updateJson = {id: res.body._id, name: '3rd party licensor new'}
+    [res, body] = yield request.putAsync({ url, json: updateJson })
+    expect(res.statusCode).toBe(200)
+    expect(res.body.name).toEqual(updateJson.name)
+    expect(res.body.creator).toEqual(user.id)
+
+describe 'GET /db/o-auth/name', ->
+  url = utils.getURL('/db/o-auth')
+  json = { name: '3rd party' }
+  beforeEach utils.wrap ->
+    yield utils.clearModels([OAuthProvider])
+
+  it 'allows licensor to update an OAuthProvider', utils.wrap ->
```

Comment:
The description says it updates the OAuthProvider, should be something like 'allows licensor to retrieve an OAuthProvider'

Reply:
Good catch, will update.